### DISCUSSION TITLE
About dealing with ghost particle

### DISCUSSION THREAD

**Discussion by yuantangjin** [rubensamarojr/polymps#4] [Q&A]
https://github.com/rubensamarojr/polymps/blob/fa0f3ca705e8c2023a9d3ae1ac2e2f1e2048a9f3/src/MpsParticle.cpp#L1356
I would like to ask, what is the meaning of the following processing in the code? 
Can it be understood that all particles beyond the scope (ie Ghost particles) are artificially gathered at the corner of the largest coordinate of the scope
```
			// Set maximum position to lastParticle
			pos[iLastParticle*3  ] = domainMaxX - partDist;
			pos[iLastParticle*3+1] = domainMaxY - partDist;
			if (dim == 2)
				pos[iLastParticle*3+2] = 0.0;
			else
				pos[iLastParticle*3+2] = domainMaxZ - partDist;
```

**Comment by rubensamarojr** [comment]
Hi There,

Exactly what you said. Deleted ghost particles are placed in the upper corner of the starting domain.
Anyway, ghost particles are disregarded in the calculations. However, to avoid some false interaction problem (a code bug that ghost particle is considered), I did this number trick.
I'm not sure if we might have a problem in any specific situation.
